server/command.go

```diff
@@ -174,6 +174,84 @@ func (p *Plugin) executeCommandQueue(args *model.CommandArgs) *model.CommandResp
 	return &model.CommandResponse{}
 }
 
+func calculateQueItemNumber(args *model.CommandArgs, p *Plugin, hashtag string) (*model.CommandResponse, int) {
+	searchResults, appErr := p.API.SearchPostsInTeamForUser(args.TeamId, args.UserId, model.SearchParameter{Terms: &hashtag})
+	if appErr != nil {
+		return responsef("Error calculating list number"), 0
+	}
+	postList := *searchResults.PostList
+	numQueueItems := len(postList.Posts)
+	return nil, numQueueItems + 1
+}
+
+func (p *Plugin) executeCommandReQueue(args *model.CommandArgs) *model.CommandResponse {
+	split := strings.Fields(args.Command)
+
+	if len(split) <= 2 {
+		return responsef("Missing parameters for requeue command")
+	}
+
+	meeting, err := p.GetMeeting(args.ChannelId)
+	if err != nil {
+		return responsef("Can't find the meeting")
+	}
+
+	oldPostID := split[2]
+	postToBeReQueued, _ := p.API.GetPost(oldPostID)
+	var (
+		prefix            string
+		hashtagDateFormat string
+	)
+	if matchGroups := meetingDateFormatRegex.FindStringSubmatch(meeting.HashtagFormat); len(matchGroups) == 4 {
+		prefix = matchGroups[1]
+		hashtagDateFormat = strings.TrimSpace(matchGroups[2])
+	}
+
+	var (
+		messageRegexFormat = regexp.MustCompile(fmt.Sprintf(`(?m)^#### #%s(?P<date>.*) [0-9]+\) (?P<message>.*)?$`, prefix))
+	)
+
+	if matchGroups := messageRegexFormat.FindStringSubmatch(postToBeReQueued.Message); len(matchGroups) == 3 {
+		originalPostDate := strings.ReplaceAll(strings.TrimSpace(matchGroups[1]), "_", " ") // reverse what we do to make it a valid hashtag
+		originalPostMessage := strings.TrimSpace(matchGroups[2])
+
+		today := time.Now()
+		local, _ := time.LoadLocation("Local")
+		formattedDate, _ := time.ParseInLocation(hashtagDateFormat, originalPostDate, local)
+		if formattedDate.Year() == 0 {
+			thisYear := today.Year()
+			formattedDate = formattedDate.AddDate(thisYear, 0, 0)
+		}
+
+		if today.Year() <= formattedDate.Year() && today.YearDay() < formattedDate.YearDay() {
+			return responsef("Re-queing future items are not supported.")
+		}
+
+		hashtag, err := p.GenerateHashtag(args.ChannelId, false, -1, true, formattedDate.Weekday())
+		if err != nil {
+			return responsef("Error calculating hashtags. Check the meeting settings for this channel.")
+		}
+
+		itemErr, numQueueItems := calculateQueItemNumber(args, p, hashtag)
+		if itemErr != nil {
+			return itemErr
```

Comment:
```suggestion
			return errors.Wrap(itemErr, "failed to calculate queue item number")
```

Reply:
Our plugin doesn't support returning this as HTTP response.

Reply:
I think the problem is that `itemErr` seems like it should be an `error` but it's a `*model.CommandResponse`. Maybe we can rename the variable `commandResponse`?